cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -121,18 +121,16 @@ macro (build_addon target prefix libs)
             foreach(src_file ${USED_SOURCES})
               file(STRINGS ${src_file} BIN_ADDON_SRC_PARTS)
               foreach(loop_var ${BIN_ADDON_SRC_PARTS})
-                string(FIND "${loop_var}" "#include" matchres)
-                if("${matchres}" EQUAL 0)
-                  string(REPLACE " " ";" loop_var "${loop_var}")
-                  list(GET loop_var 1 include_name)
-                  string(REGEX REPLACE "[<>\"]|kodi/" "" include_name "${include_name}")
-                  if(include_name MATCHES ${depend_header})
+                string(REGEX MATCH "^[ \t]*#[ \t]*(include|import)[ \t]*[\"<].*[\">]" loop_var "${loop_var}")
+                if(loop_var AND loop_var MATCHES "[\"<]kodi/${depend_header}[\">]")
+                  get_directory_property(CURRENT_DEFS COMPILE_DEFINITIONS)
+                  if(NOT used_type_name IN_LIST CURRENT_DEFS)
                     set(ADDON_DEPENDS "${ADDON_DEPENDS}\n<import addon=\"${${xml_entry_name}}\" minversion=\"${${depends_minver}}\" version=\"${${depends_ver}}\"/>")
-                    # Inform with them the addon header about used type
+                    # Inform with them the addon header about used type, if not present before
                     add_definitions(-D${used_type_name})
                     message(STATUS "Added usage definition: ${used_type_name}")
-                    set(FOUND_HEADER_USAGE 1)
                   endif()
+                  set(FOUND_HEADER_USAGE 1)
```

Comment:
Why move setting `FOUND_HEADER_USAGE`?

Reply:
Is not moved, only a bit bad presented from git, look the two patches alone and you see. But I move it up to mesage, let see what come.